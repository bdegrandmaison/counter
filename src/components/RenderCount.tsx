type RenderCountProp = {
    count: number
}

export default function RenderCount( {count}: RenderCountProp) {
    return <p>{count}</p>
}