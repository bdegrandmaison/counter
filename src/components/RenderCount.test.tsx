import { render, screen } from "@testing-library/react";
import RenderCount from './RenderCount'

describe('Count display component', () => {
    it.each`
        count | expectedValue
        ${10} | ${'10'}
        ${20} | ${'20'}
        ${30} | ${'30'}
        ${40} | ${'40'}
    
    `('should render the value $expectedValue given in the count prop "$count"', ({ count, expectedValue }) => {
        const { container: element } = render(<RenderCount count={count} />)
    
        expect(screen.getByText(expectedValue)).toBeInTheDocument()
        expect(element).toMatchSnapshot();
    })
});
