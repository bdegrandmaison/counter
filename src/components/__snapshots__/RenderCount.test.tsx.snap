// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Count display component should render the value 10 given in the count prop "10" 1`] = `
<div>
  <p>
    10
  </p>
</div>
`;

exports[`Count display component should render the value 20 given in the count prop "20" 1`] = `
<div>
  <p>
    20
  </p>
</div>
`;

exports[`Count display component should render the value 30 given in the count prop "30" 1`] = `
<div>
  <p>
    30
  </p>
</div>
`;

exports[`Count display component should render the value 40 given in the count prop "40" 1`] = `
<div>
  <p>
    40
  </p>
</div>
`;
